src/schema/term.rs

```diff
@@ -156,18 +166,28 @@ impl Term {
         self.set_bytes(text.as_bytes());
     }
 
-    /// Removes the value_bytes and set the type code.
-    pub fn clear_with_type(&mut self, typ: Type) {
-        self.truncate(5);
-        self.0[4] = typ.to_code();
+    /// Truncates the term. The new length needs to be at least 5, which is reserved for metadata.
+    fn truncate(&mut self, len: usize) {
+        assert!(len >= TERM_METADATA_LENGTH);
+        self.0.truncate(len);
```

Comment:
this is not fixed. `len` means something different than in other place of the code right now.

Reply:
Is this method necessary, or can we leave with just truncate_value_bytes

Reply:
yes I replaced it with `truncate_value_bytes(0)`